src/main/js/dynamic/DistributionConfiguration.js

```diff
@@ -107,66 +106,73 @@ class DistributionConfiguration extends Component {
             const keepFrequency = Object.assign(keepName, FieldModelUtilities.updateFieldModelSingleValue(updatedChannelConfig, KEY_FREQUENCY, frequency));
             const keepProvider = Object.assign(keepFrequency, FieldModelUtilities.updateFieldModelSingleValue(updatedChannelConfig, KEY_PROVIDER_NAME, provider));
 
-            this.props.checkDescriptorForGlobalConfig(KEY_CHANNEL_NAME, newChannel.name);
+            validateDescriptorForGlobalConfig(KEY_CHANNEL_NAME, newChannel.name);
             this.setState({
                 channelConfig: keepProvider,
                 currentChannel: newChannel
             });
         }
-        const selectedProviderOption = FieldModelUtilities.getFieldModelSingleValue(this.state.channelConfig, KEY_PROVIDER_NAME);
+        const selectedProviderOption = FieldModelUtilities.getFieldModelSingleValue(channelConfig, KEY_PROVIDER_NAME);
         const prevProviderName = currentProvider ? currentProvider.name : '';
 
         if (selectedProviderOption && prevProviderName !== selectedProviderOption) {
-            const newProvider = this.props.descriptors.find((descriptor) => descriptor.name === selectedProviderOption && descriptor.context === DescriptorUtilities.CONTEXT_TYPE.DISTRIBUTION);
+            const newProvider = descriptors.find((descriptor) => descriptor.name === selectedProviderOption && descriptor.context === DescriptorUtilities.CONTEXT_TYPE.DISTRIBUTION);
             const emptyProviderConfig = FieldModelUtilities.createFieldModelWithDefaults(newProvider.fields, newProvider.context, newProvider.name);
-            this.props.checkDescriptorForGlobalConfig(KEY_PROVIDER_NAME, newProvider.name);
+            validateDescriptorForGlobalConfig(KEY_PROVIDER_NAME, newProvider.name);
             this.setState({
                 providerConfig: emptyProviderConfig,
                 currentProvider: newProvider
             });
         }
+
+        if (prevProps.status === 'VALIDATING' && status === 'VALIDATED') {
+            const jsonBody = this.buildJsonBody();
+            if (isUpdatingJob) {
+                updateDistribution(jsonBody);
+            } else {
+                saveDistribution(jsonBody);
+            }
+        }
     }
 
-    handleClose() {
-        this.setState({ show: false });
-        this.props.onModalClose();
-        this.props.handleCancel();
+    setSendMessageVisible(visible) {
+        this.setState({
+            showSendMessage: visible
+        });
+    }
+
+    handleTestSubmit(event) {
+        event.preventDefault();
+        const { testDistribution } = this.props;
+        const jsonBody = this.buildJsonBody();
+        testDistribution(jsonBody);
     }
 
     buildJsonBody() {
         const { channelConfig, providerConfig } = this.state;
+        const { jobId } = this.props;
```

Comment:
Moved the code that checked the job id into this function instead of having the check in multiple places.